From: rofl0r <rofl0r@users.noreply.github.com>
Date: Mon, 15 Nov 2021 19:00:56 +0000
Subject: [PATCH] linux: always accept classic joystick regardless of whether
 hint is set

the hint in question is SDL_LINUX_JOYSTICK_CLASSIC.

--- a/src/joystick/linux/SDL_sysjoystick.c
+++ b/src/joystick/linux/SDL_sysjoystick.c
@@ -297,15 +297,6 @@
         if (!(udev_class & SDL_UDEV_DEVICE_JOYSTICK)) {
             return;
         }
-        if (SDL_classic_joysticks) {
-            if (!IsJoystickJSNode(devpath)) {
-                return;
-            }
-        } else {
-            if (IsJoystickJSNode(devpath)) {
-                return;
-            }
-        }
 
         /* Wait a bit for the hidraw udev node to initialize */
         SDL_Delay(10);
@@ -561,11 +552,7 @@
 
 static SDL_bool IsJoystickDeviceNode(const char *node)
 {
-    if (SDL_classic_joysticks) {
-        return IsJoystickJSNode(node);
-    } else {
-        return IsJoystickEventNode(node);
-    }
+    return IsJoystickEventNode(node) || IsJoystickJSNode(node);
 }
 
 #ifdef HAVE_INOTIFY
